src/Raven.Pal/src/posix/spawn.c

```diff
@@ -0,0 +1,326 @@
+#ifndef _GNU_SOURCE
+#define _GNU_SOURCE
+#endif
+
+#include <sys/wait.h>
+#include <pthread.h>
+#include <sys/time.h>
+#include <unistd.h>
+#include <stdlib.h>
+#include <stdio.h>
+#include <sys/types.h>
+#include <sys/stat.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <sys/mman.h>
+#include <assert.h>
+#include <time.h>
+#include <string.h>
+#include <signal.h>
+#include <dirent.h>
+
+#include "rvn.h"
+#include "status_codes.h"
+#include "internal_posix.h"
+
+PRIVATE char** 
+_parse_cmd_line(char* line, char *filename)
+{
+    int argc = 2;
+    char** argv = malloc(argc);
+    argv[0] = filename;
+    argv[--argc] = NULL;    
+    if (line == NULL || *line == '\0')
+        return argv;
+    
+    char *rest = line;
+    char *token;
+
+    while((token = strtok_r(rest, " \t\n", &rest)))
+    {
+        argv[argc++] = token;
+        argv = realloc(argv, argc + 1);
```

Comment:
Check if this failed

Reply:
argv = realloc(argv, argc + 1); - see comment above